lib/simulator_basic.h

```diff
@@ -853,6 +901,457 @@ class SimulatorBasic final {
              state.num_qubits(), cmaskh, emaskh, rstate);
   }
 
+  std::complex<double> ExpectationValue1H(const std::vector<unsigned>& qs,
```

Comment:
These `ExpectationValue*H` methods are very similar to each other. Would combining them negatively impact performance? (Integer template values may help with this)

Reply:
Combining them by using a template with integer template parameters should not impact performance.
However, these methods are generated. So I think it is not a problem if there is a lot of similar code. Anyway combining ExpectationValue* (and ApplyGate*) is a quite involved task and it should not be a part of this PR.